api/apps/geoprocessing/src/modules/planning-units/planning-units.service.ts

```diff
@@ -40,47 +41,88 @@ export class PlanningUnitsService {
   private readonly logger: Logger = new Logger(PlanningUnitsService.name);
   constructor(
     @InjectRepository(PlanningUnitsGeom)
-    private readonly protectedAreasRepository: Repository<PlanningUnitsGeom>,
+    private readonly planningUnitsRepository: Repository<PlanningUnitsGeom>,
     @Inject(TileService)
     private readonly tileService: TileService,
   ) {}
 
   /**
-   * @param bbox boundingbox of the area where the grids would be generated
    * @param planningUnitGridShape the grid shape that would be use for generating the grid. This grid shape
    * can be square or hexagon. If any grid shape is provided, square would be the default.
    * @param planningUnitAreakm2 area in km2 of the individual grid that would be generated.
    * If any value is not provided, 4000 would be the default.
    */
+  regularFuncionGridSelector(
+    planningUnitGridShape: PlanningUnitGridShape
+  ): string {
+    const functEquivalence: { [key in keyof typeof PlanningUnitGridShape]: string} = {
+    hexagon: 'ST_HexagonGrid',
+    square: 'ST_SquareGrid',
+    }
 
-  buildPlanningUnitsWhereQuery(
+    return functEquivalence[planningUnitGridShape]
+  }
+
+  calculateGridSize(
+    planningUnitGridShape: PlanningUnitGridShape,
+    planningUnitAreakm2: number
+    ): number {
+
+    return Math.sqrt(planningUnitAreakm2) * 1000
+  }
+  /**
+   * @param bbox bounding box of the area where the grids would be generated
+   * @param planningUnitGridShape the grid shape that would be use for generating the grid. This grid shape
+   * can be square or hexagon. If any grid shape is provided, square would be the default.
+   * @param planningUnitAreakm2 area in km2 of the individual grid that would be generated.
+   * If any value is not provided, 4000 would be the default.
+   */
+  buildPlanningUnitsCustomQuery(
     x: number,
     y: number,
     z: number,
     planningUnitGridShape: PlanningUnitGridShape,
     planningUnitAreakm2: number,
     filters?: PlanningUnitsFilters,
   ): string {
-    let PlanningUnitGridShape = '';
-    let whereQuery = '';
-    if (planningUnitGridShape == 'hexagon') {
-      PlanningUnitGridShape = 'ST_HexagonGrid';
-    }
-    if (planningUnitGridShape == 'square') {
-      PlanningUnitGridShape = 'ST_SquareGrid';
+    const gridShape = this.regularFuncionGridSelector(planningUnitGridShape)
+    const gridSize = this.calculateGridSize(planningUnitGridShape,
+      planningUnitAreakm2)
+    const ratioPixelExtent = (gridSize / (156412/(2**z)))
+    let Query = `( SELECT row_number() over() as id, (${gridShape}(${gridSize}, \
```

Comment:
with capital letter it suggests it's a class, I'd use `query`

Reply:
fixed